src/scripts/validate-source-files.ts

```diff
@@ -140,137 +127,130 @@ function error(...params: any[]) {
   console.error(...params.map((p) => chalk.red(p)))
 }
 
-const DISTRICTS = `${ADMIN_STRUCTURE_SOURCE}source/districts.csv`
-const STATISTICS = `${ADMIN_STRUCTURE_SOURCE}source/statistics.csv`
-const STATES = `${ADMIN_STRUCTURE_SOURCE}source/states.csv`
-const EMPLOYEES = `${EMPLOYEES_SOURCE}source/test-employees.csv`
-const CRVS_FACILITIES = `${FACILITIES_SOURCE}source/crvs-facilities.csv`
-const HEALTH_FACILITIES = `${FACILITIES_SOURCE}source/health-facilities.csv`
-
 async function main() {
-  log(chalk.yellow('Validating file:'), chalk.bold(STATES))
-  const rawStates = await readCSVToJSON(STATES)
-  const states = Location.array().parse(rawStates)
-  log(chalk.green('File is valid'), '✅', '\n')
-
-  log(chalk.yellow('Validating file:'), chalk.bold(DISTRICTS))
-
-  const rawDistricts = await readCSVToJSON(DISTRICTS)
-
-  const districts = Location.array().parse(rawDistricts)
+  log(chalk.yellow('Validating locations file.'))
+  const rawLocations: [] = await readCSVToJSON(process.argv[2])
+  const countryExpression = /^admin0name$/i
+  const nameExpression = /^admin(\d+)name_en$/i
+  const aliasExpression = /^admin(\d+)name_alias$/i
+  const pcodeExpression = /^admin(\d+)Pcode$/i
+  const csvLocationHeaders = [...new Set(rawLocations.flatMap(Object.keys))]
+  csvLocationHeaders.map((header: string) => {
+    let headersOK: RegExpExecArray | null
+    headersOK = countryExpression.exec(header)
+    if (!headersOK) {
+      headersOK = nameExpression.exec(header)
+      if (!headersOK) {
+        headersOK = aliasExpression.exec(header)
+        if (!headersOK) {
+          headersOK = pcodeExpression.exec(header)
+          if (!headersOK) {
+            error(
+              chalk.blue(`LOCATIONS ERROR!:`),
+              chalk.yellow(`Header ${header}`),
+              'is not a supported header'
+            )
+            process.exit(1)
+          }
+        }
+      }
+    }
+  })
+  let ADMIN_LEVELS = 0
 
-  for (const district of districts) {
-    const state = states.find(
-      (state) => district.partOf === `Location/${state.statisticalID}`
-    )
-    if (!state) {
-      error(
-        DISTRICTS,
-        `Line ${districts.indexOf(district) + 2}`,
-        'District',
-        district.name,
-        'is not part of any known state'
-      )
-      process.exit(1)
+  csvLocationHeaders.find((header: string) => {
+    const currentLevel = /^admin(\d+)pcode/i.exec(header)
+    if (currentLevel) {
+      ADMIN_LEVELS < Number(currentLevel[1])
+        ? (ADMIN_LEVELS = Number(currentLevel[1]))
+        : ADMIN_LEVELS
     }
-  }
-  log(chalk.green('File is valid'), '✅', '\n')
-  log(chalk.yellow('Validating file:'), chalk.bold(STATISTICS))
-  const rawStatistics = await readCSVToJSON(STATISTICS)
+  })
+  log(chalk.green('ADMIN_LEVELS parsed'), '✅', '\n')
 
+  log(chalk.yellow('Validating statistics file.'))
+  const rawStatistics = await readCSVToJSON(process.argv[6])
```

Comment:
We need to make sure that a statistic exists for every single location.  So we have the location list in this file "rawLocations", we need to make sure that a statistic row exists for every one.  We also need to make sure that there are no duplicate locations or statistics.  Same applies to all files we are important.  If there are any missing rows, we would error.